src/exportMap/childContext.js

```diff
@@ -17,13 +25,28 @@ export default function childContext(path, context) {
     prevSettings = JSON.stringify(settings);
   }
 
-  if (JSON.stringify(parserOptions) !== prevParserOptions) {
-    parserOptionsHash = hashObject({ parserOptions }).digest('hex');
-    prevParserOptions = JSON.stringify(parserOptions);
+  // We'll use either a combination of `parserOptions` and `parserPath` or `languageOptions`
+  // to construct the cache key, depending on whether this is using a flat config or not.
+  let optionsToken;
+  if (!parserPath && languageOptions) {
+    if (JSON.stringify(languageOptions, stringifyReplacerFn) !== prevOptions) {
+      optionsHash = hashObject({ languageOptions }).digest('hex');
+      prevOptions = JSON.stringify(languageOptions, stringifyReplacerFn);
+    }
+    // For languageOptions, we're just using the hashed options as the options token
+    optionsToken = optionsHash;
+  } else {
+    if (JSON.stringify(parserOptions) !== prevOptions) {
+      optionsHash = hashObject({ parserOptions }).digest('hex');
+      prevOptions = JSON.stringify(parserOptions);
+    }
+    // When not using flat config, we use a combination of the hashed parserOptions
+    // and parserPath as the token
+    optionsToken = String(parserPath) + optionsHash;
   }
 
   return {
-    cacheKey: String(parserPath) + parserOptionsHash + settingsHash + String(path),
+    cacheKey: optionsToken + settingsHash + String(path),
```

Comment:
would it make more sense to just hash a single object composed of all these pieces, rather than multiple hashes?

Reply:
We could. I think the one downside is that if only settings change or if only one of the options changes, we'd have to generate a hash for a larger object, rather than hashing just settings or just the options object (the part of it that changed).  So, two hashes initially and smaller incremental hashes as things change, vs one bigger hash that would have to be regenerated when any portion of it changes.  Which option would be more performant would likely depend on how we expect people to interact with the cache.

Reply:
that's a solid argument.